src/pins/gen.rs

```diff
@@ -52,12 +52,32 @@ macro_rules! pins {
             ///
             /// [`Pin`]: struct.Pin.html
             #[allow(non_camel_case_types)]
-            pub struct $type(());
+            pub struct $type {
+                PORT: usize,
+                ID  : u8,
+                MASK: u32,
+            }
```

Comment:
I think this is fine for a first version, but not ideal. We know the correct values of these for `$type` and can just return those values from the methods directly. No need to save them in the struct, which is uncessary overhead.

Also, as a general note: I prefer established Rust naming convention, so `port`, `id`, and `mask`. This crate does have a lot of all-caps names, but they usually refer to names that are the same in the user manual.

Reply:
ah right, that makes sense. Will change!